ddtrace/internal/ci_visibility/encoder.py

```diff
@@ -96,20 +97,25 @@ def _build_payload(self, traces):
             if (is_not_xdist_worker or span.get_tag(EVENT_TYPE) != SESSION_TYPE)
         ]
         if not normalized_spans:
-            return None
+            return []
         record_endpoint_payload_events_count(endpoint=ENDPOINT.TEST_CYCLE, count=len(normalized_spans))
 
         # TODO: Split the events in several payloads as needed to avoid hitting the intake's maximum payload size.
-        return CIVisibilityEncoderV01._pack_payload(
-            {"version": self.PAYLOAD_FORMAT_VERSION, "metadata": self._metadata, "events": normalized_spans}
-        )
+        return [
+            (
+                CIVisibilityEncoderV01._pack_payload(
+                    {"version": self.PAYLOAD_FORMAT_VERSION, "metadata": self._metadata, "events": normalized_spans}
+                ),
+                len(traces),
+            )
+        ]
 
     @staticmethod
     def _pack_payload(payload):
         return msgpack_packb(payload)
 
     def _convert_span(self, span, dd_origin, new_parent_session_span_id=0):
-        # type: (Span, str, Optional[int]) -> Dict[str, Any]
+        # type: (Span, Optional[str], Optional[int]) -> Dict[str, Any]
```

Comment:
nit: We can start using python3 types

Reply:
I can do that in a follow up PR (It's a 2-parts PR actually)